cardano-crypto-class/src/Cardano/Crypto/Schnorr.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE CApiFFI #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Crypto.Schnorr (
```

Comment:
Hmmm, do you think we could contribute this upstream to `secp256k1-haskell`?

Reply:
We could, but I chose not to, in the interests of getting this out the door faster. Possibly useful for the future though.